plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtils.java

```diff
@@ -35,22 +34,22 @@
   private static final Logger logger = Logger.getLogger(ModuleUtils.class.getName());
 
   /**
-   * Retrieve the &lt;service&gt; or &lt;module&gt; identifier from the <tt>appengine-web.xml</tt>.
-   * If an identifier is not found, or <tt>appengine-web.xml</tt> is not found, then returns
-   * "default".
+   * Retrieve the &lt;service&gt; or &lt;module&gt; identifier from <tt>appengine-web.xml</tt>.
+   * If an identifier is not found, then returns "default".
+   * If <tt>appengine-web.xml</tt> is not found, then returns null.
```

Comment:
The original implementation was wrong, it is true.  My intent was for this method to be `@NonNull`.

But my choice here, to return _"default"_ if the file didn't exist, was deliberate: if you want the untrammelled information, use `AppEngineDescriptor` directly.  In practice our `canModify()` first verifies all modules have an App Engine facet, which should ensure they have an `appengine-web.xml`.

If you really think we should return `null` here, then we need to fix up `LocalAppEngineServerDelegate` too.